sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_service_client.hpp

```diff
@@ -67,15 +67,14 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     std::string GetUrl() const { return m_serviceUrl.GetAbsoluteUrl(); }
 
     /**
-     * @brief List the shares from the service.
+     * @brief Returns a sequence of the shares in the storage account. Enumerating the shares may
+     * make multiple requests to the service while fetching all the values.
      * @param options Optional parameters to list the shares.
      * @param context Context for cancelling long running operations.
-     * @return Azure::Response<Models::ListSharesSinglePageResult> The results containing the
-     * shares returned and information used for future list operation on valid result not yet
-     * returned.
+     * @return ListSharesPagedResponse describing the shares in this storage account.
      */
-    Azure::Response<Models::ListSharesSinglePageResult> ListSharesSinglePage(
-        const ListSharesSinglePageOptions& options = ListSharesSinglePageOptions(),
+    ListSharesPagedResponse ListShares(
```

Comment:
😂😂😂 If we add iterator in the future, it'll be exactly the same as my PR https://github.com/Azure/azure-sdk-for-cpp/pull/2092, which was just rejected in the meeting yesterday.

Reply:
Yes, this was considered with @CaseyCarter. Casey turned me on to the statis inheritance thing. And I think we should turn Response<T> into the static inheritance thing because we don't need/want the inheritance/polymorphism. 
I just didn't want to bring this up so close to shipping.

Reply:
> I think we should turn Response into the static inheritanc

What's the benefit?

> because we don't need/want the inheritance/polymorphism.

Why don't we want this?

Reply:
Is-A is useful for polymorphism. What is the case where we need to treat multiple response types as a generic (less capable) Response<T>?